Linux About Modal

jfgordon2 opened the pull request:
## Description

Updated the "About GitHub Desktop" model to remove the button to check for updates (since it didn't do anything on Linux) and replaced with a link to the linux releases page

### Screenshots

![image](https://user-images.githubusercontent.com/55799997/77884732-44b38f80-722b-11ea-90a5-af7f6da323a1.png)

## Release notes

Notes: Removed the 'check for updates' button, and replaced with a link to the unofficial linux releases page.

jfgordon2 committed:
Linux About Modal

Updated the "About GitHub Desktop" model to remove the button to check for updates (since it didn't do anything on Linux) and replace with a link to the linux releases page

shiftkey commented on app/src/ui/about/about.tsx:
```diff
@@ -94,6 +95,17 @@ export class About extends React.Component<IAboutProps, IAboutState> {
 
+    if (__LINUX__) {
+      const linuxReleaseLink = (
+        <LinkButton uri={LinuxReleasesUri}>View Linux Releases</LinkButton>
```
I think we can drop this word to keep it simple:

```suggestion
        <LinkButton uri={LinuxReleasesUri}>View Releases</LinkButton>
```

jfgordon2 replied:
@shiftkey I've updated this in the new commit - thanks!

shiftkey commented on app/src/ui/about/about.tsx:
```diff
@@ -15,6 +15,7 @@ import { Loading } from '../lib/loading'
 import { RelativeTime } from '../relative-time'
 import { assertNever } from '../../lib/fatal-error'
 import { ReleaseNotesUri } from '../lib/releases'
+import { LinuxReleasesUri } from '../lib/releases'
```
:art: this can be combined with the previous `import`

jfgordon2 replied:
@shiftkey - I've updated this in the new commit.  Thanks for the guidance and feedback!

shiftkey reviewed (commented):
A couple of suggestions to tidy this up, otherwise this is :gem:

jfgordon2 committed:
Clean Up

Updated to consolidate import and removed the word "Linux" from the release link

shiftkey reviewed (approved):
Thanks for the polish here @jfgordon2!

sarim commented:
Not sure what is happening here ^^^.  But Maybe a simple update checker could be added by checking github release? ex: `https://api.github.com/repos/shiftkey/desktop/releases?per_page=1`

jfgordon2 commented:
> Not sure what is happening here ^^^.  But Maybe a simple update checker could be added by checking github release? ex: `https://api.github.com/repos/shiftkey/desktop/releases?per_page=1`

Hey @sarim! I did this forever ago just to kind of improve my own experience on linux.  Since then, @shiftkey and others have done a huge amount of work bringing it to packagecloud (for deb and rpm) as well as [flatpak](https://flathub.org/apps/details/io.github.shiftey.Desktop) which, at least in my opinion, provide a much better and more integrated update/install experience on the linux desktop than linking to this repo.  Honestly, even just still having this link might be misdirected when presented with these superior options.  Hopefully those resources might help you to keep this great app up to date!
